### PR TITLE
Add docs for contributors.

### PR DESCRIPTION
**Pull request by barrachri**
This PR try to address https://github.com/ray-project/ray/issues/1185.

Adding also github issue and PR templates. 

It's WIP because I'd like to improve/clarify:

* testing process
* how to get a dev env ready
* how to collect new contributors

### CODE REVIEW

**Comment by robertnishihara** [comment]
ok to test

**Comment by robertnishihara** [comment]
Thanks for getting this started, this'll be really helpful!

I haven't seen any other projects use a pull request template. Do you know of any?

The CONTRIBUTING.rst file could be modeled after https://github.com/apache/arrow/blob/master/.github/CONTRIBUTING.md as well as https://github.com/tensorflow/tensorflow/blob/master/CONTRIBUTING.md. It may make sense to put instructions for getting set up for development somewhere else in the documentation. I don't feel strongly about this though.

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+   $ pip install -r requirements-doc.txt
+   $ make html
+
+Once it finishes it will output the index html page
+``open file:///.../ray/docs/_build/html/index.html``.
+
+Go to the link and make sure your doc changes looks good.
+
+
+Changelog update
```
Comment (robertnishihara) [inline CONTRIBUTING.rst]: This section isn't relevant, is it?

**Comment by barrachri** [comment]
About the pull request template just to cite a few:

https://github.com/pandas-dev/pandas/blob/master/.github/PULL_REQUEST_TEMPLATE.md
https://github.com/dask/dask/blob/master/.github/PULL_REQUEST_TEMPLATE.md

Also [Keras](https://github.com/fchollet/keras/blob/master/CONTRIBUTING.md) has also something similar but inside the CONTRIBUTING.md

I think it's nice to have because you can see what is needed directly inside the PR.
In this way as a admin you don't need to always remember that something is missing.

File: CONTRIBUTING.rst (hunk 1 above)
Comment (barrachri) [reply CONTRIBUTING.rst]: If you send a PR about improving the docs I think it's a great idea the check the docs (so open the local link and...).

About Changelog, how do you collect the changes so far?
This is really useful if you want to collect info between releases and you know what changed.

Comment (robertnishihara) [reply CONTRIBUTING.rst]: We currently don't maintain a changelog. We could certainly add one, though we can always construct it after the fact just by looking through the commit history.

Comment (barrachri) [reply CONTRIBUTING.rst]: Ok, so I will remove this changelog part.

**Comment by robertnishihara** [comment]
Thanks for the links, those look reasonable.

**Comment by robertnishihara** [comment]
@barrachri Mind if I take a pass over these docs (e.g., and push to your branch)? I'd prefer to start with more minimal versions of these files and then add to them as appropriate.

**Comment by barrachri** [comment]
@robertnishihara you are more than welcome in doing so.

**Comment by robertnishihara** [comment]
@barrachri I removed a lot of stuff, want to take a look? I'd suggest that we get this merged and then augment it as needed in follow up PRs.

**Comment by robertnishihara** [comment]
I took some of the issue template from TensorFlow and the contributing doc from Arrow.

**Comment by barrachri** [comment]
Green light, I think you can merge!

**Review by pcmoritz — APPROVED** [review]

**Comment by robertnishihara** [comment]
Thanks @barrachri!

**Pull request merged**